### ISSUE TITLE
Vector swizzling

### ISSUE THREAD

**Issue by davidson16807** [openscad/openscad#912]
[Swizzling](http://en.wikipedia.org/wiki/Swizzling_%28computer_graphics%29) is a useful concept you can find built in to computer graphics languages like [GLSL](http://en.wikipedia.org/wiki/OpenGL_Shading_Language). There's been more than one occasion where I would have liked to use it. It can be very useful for converting 3d vectors to 2d and rearranging elements of a vector.

To pose an example, the following statements would produce equivalent behavior:

```
foo3d = [1,2,3];

foo2d = [foo3d.x, foo3d.y];  // without swizzling
foo2d = foo3d.xy;            // with swizzling

bar = [foo3d.z, foo3d.y, foo3d.x];
bar = foo3d.zyx;

baz = [foo3d.x, foo3d.x, foo3d.x, foo3d.x];
baz = foo3d.xxxx;
```

This easily handles conversion to lower dimensionality, though it wouldn't support all scenarios when converting to higher dimensionality. The latter is particularly important because it allows you to convert vectors to a form that is compatible with 4d matrices. As an optional extension to this proposal, I suggest a method be found to convert to higher dimensionality, perhaps borrowing GLSL's vec*() functions:

```
baz = [foo3d.x, foo3d.y, foo3d.z, 1];
baz = vec4(foo3d, 1);
```

or Python's asterisk notation:

```
baz = [foo3d.x, foo3d.y, foo3d.z, 1];
baz = [*foo3d, 1];
```
##

**Comment by TLC123** [comment]
I keep a bunch of swizzling  functions at hand. Like so:


function xyy(v)=[v.x,.v.y,v.y];
function zyx(v)=[v.z,.v.y,v.xy];
function xyzw(v)=[v.x,.v.y,v.y,v[3]];
function w(v)=v[3];
function rgg(v)=[v.x,.v.y,v.y];

**Comment by t-paul** [comment]
Closed via #4010.
